debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcDatabaseHistory.java

```diff
@@ -0,0 +1,278 @@
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Splitter;
+
+import io.debezium.annotation.ThreadSafe;
+import io.debezium.common.annotation.Incubating;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.document.DocumentReader;
+import io.debezium.document.DocumentWriter;
+import io.debezium.relational.history.AbstractDatabaseHistory;
+import io.debezium.relational.history.DatabaseHistory;
+import io.debezium.relational.history.DatabaseHistoryException;
+import io.debezium.relational.history.DatabaseHistoryListener;
+import io.debezium.relational.history.HistoryRecord;
+import io.debezium.relational.history.HistoryRecordComparator;
+import io.debezium.util.Collect;
+import io.debezium.util.FunctionalReadWriteLock;
+
+/**
+ * A {@link DatabaseHistory} implementation that stores the schema history to database table
+ *
+ * @author Ismail Simsek
+ */
+@ThreadSafe
+@Incubating
+public final class JdbcDatabaseHistory extends AbstractDatabaseHistory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JdbcDatabaseHistory.class);
+
+    public static final Field JDBC_URI = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.uri")
+            .withDescription("Uri of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+
+    public static final Field JDBC_USER = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.user")
+            .withDescription("Username of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+
+    public static final Field JDBC_PASSWORD = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.password")
+            .withDescription("Password of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+
+    private static final String DATABASE_HISTORY_TABLE_NAME = "debezium_database_history";
+
+    private static final String DATABASE_HISTORY_TABLE_DDL = "CREATE TABLE " + DATABASE_HISTORY_TABLE_NAME +
+            "(" +
+            "id VARCHAR(36) NOT NULL," +
+            "history_data VARCHAR(65000)," +
+            "history_data_seq INTEGER," +
+            "record_insert_ts TIMESTAMP NOT NULL," +
+            "record_insert_seq INTEGER NOT NULL" +
+            ")";
+
+    private static final String DATABASE_HISTORY_TABLE_SELECT = "SELECT id, history_data, history_data_seq FROM " + DATABASE_HISTORY_TABLE_NAME
+            + " ORDER BY record_insert_ts, record_insert_seq, id, history_data_seq";
+
+    private static final String DATABASE_HISTORY_TABLE_DATA_EXISTS_SELECT = "SELECT * FROM " + DATABASE_HISTORY_TABLE_NAME + " LIMIT 1";
```

Comment:
Again issue with SQL dialect. This is form example for SQL Server but Oracle or Postgres would use a different statement.